src/Amadevus.RecordGenerator.Generators/EqualityUsageAnalyzer.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Amadevus.RecordGenerator.Analyzers;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Amadevus.RecordGenerator.Generators
+{
+    internal sealed class EqualityUsageAnalyzer
+    {
+        public static IEnumerable<Diagnostic> GenerateDiagnostics(RecordDescriptor descriptor)
+        {
+            var sealedToken = SyntaxFactory.Token(SyntaxKind.SealedKeyword);
+            if (descriptor.TypeDeclaration.Modifiers.Contains(sealedToken)) yield break;
```

Comment:
This doesn't work. Try:

```csharp
            if (descriptor.TypeDeclaration.Modifiers.Any(token => token.IsKind(SyntaxKind.SealedKeyword))) yield break;
```

`Contains` uses default equality, and it seems those tokens are not equal almost all the time (maybe whitespace trivia is the reason). Anyway, my snippet solves the problem.

Not sure how, but this could probably do with some tests...

Reply:
You were right, I implemented it the way you suggested.

Yes, the whole feature sill needs to be tested. Tests will come as soon